descriptor/descriptor.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package descriptor
+
+import (
+	"errors"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+var ErrMissingMediaType error = errors.New("missing media type")
+
+// Generate returns a descriptor, given the content and media type.
+func Generate(content []byte, mediaType string) (ocispec.Descriptor, error) {
```

Comment:
We can call it `DescriptorFromBytes`.

Reply:
@Wwwsylvia How about
```go
func NewDescriptorFromBytes(mediaType string, content []byte) ocispec.Descriptor
```
as inspired by https://pkg.go.dev/github.com/opencontainers/go-digest#NewDigestFromBytes.

If `mediaType` is empty, we can default it to `application/octet-stream`.

Reply:
Looks good.

Reply:
Renamed the function and set the default media type when `mediaType` is empty. Removed returning an error.